marconi/src/Marconi/Indexers.hs

```diff
@@ -87,21 +84,28 @@ initialCoordinator indexerCount =
               <*> newQSemN 0
               <*> pure indexerCount
 
-type Worker = Coordinator -> TChan (ChainSyncEvent (BlockInMode CardanoMode)) -> FilePath -> IO ()
+-- Currently this `a` should stand for [ChainPoint], but I think eventually it should
+-- be constrained to a set of indexer type classes viewed through an MVar. This will
+-- allow indexers to be shared and queried safely.
+type Worker = Coordinator -> FilePath -> IO [Storable.StorablePoint ScriptTx.ScriptTxHandle]
 
 datumWorker :: Worker
-datumWorker Coordinator{_barrier} ch path = Datum.open path (Datum.Depth 2160) >>= innerLoop
+datumWorker Coordinator{_barrier, _channel} path = do
+  ix <- Datum.open path (Datum.Depth 2160)
+  workerChannel <- atomically . dupTChan $ _channel
+  void . forkIO $ innerLoop workerChannel ix
+  pure [ChainPointAtGenesis]
   where
-    innerLoop :: DatumIndex -> IO ()
-    innerLoop index = do
+    innerLoop :: TChan (ChainSyncEvent (BlockInMode CardanoMode)) -> DatumIndex -> IO ()
+    innerLoop ch index = do
```

Comment:
I don't particularly like defining functions in let loops. I also kind of like to have the variables used in a function defined as arguments rather than closures.